File: 1089-duplicate-zeros/1089-duplicate-zeros.py
class Solution:
    def duplicateZeros(self, arr: List[int]) -> None:
        """
        Do not return anything, modify arr in-place instead.
        """
        nums=[]
        i=0
        j=0
        while j<len(arr) and len(nums)<=len(arr):
            if arr[i]==0:
                nums.append(0)
                if len(nums)==len(arr):
                    break
                nums.append(0)
                j+=1
            else:
                nums.append(arr[i])
            j+=1
            i+=1
        for k in range(len(nums)):
            arr[k]=nums[k]
        

File: 1089-duplicate-zeros/1089-duplicate-zeros.js
/**
 * @param {number[]} arr
 * @return {void} Do not return anything, modify arr in-place instead.
 */
var duplicateZeros = function(arr) {
    let i=0
    let k=0
    let nums=[]
    while(k<arr.length){
        if(arr[i]===0){
            nums[k]=0
            if(nums.length===arr.length){
                break
            }
            nums[k+1]=0
            k++
        }else{
            nums[k]=arr[i]
        }
        i++
        k++
    }
    for(let j=0;j<nums.length;j++){
        arr[j]=nums[j]
    }
};